Purchases/SwiftSources/LocalReceiptParsing/ReceiptParser.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ReceiptParser.swift
+//  Purchases
+//
+//  Created by Andrés Boedo on 7/22/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+struct ReceiptParser {
+    private let objectIdentifierParser: ASN1ObjectIdentifierBuilder
+    private let containerBuilder: ASN1ContainerBuilder
+    private let receiptBuilder: AppleReceiptBuilder
+
+    init() {
+        self.objectIdentifierParser = ASN1ObjectIdentifierBuilder()
+        self.containerBuilder = ASN1ContainerBuilder()
+        self.receiptBuilder = AppleReceiptBuilder()
+    }
+
+    func parse(from data: Data) throws -> AppleReceipt {
+        let intData = [UInt8](data)
+
+        let asn1Container = try containerBuilder.build(fromPayload: ArraySlice(intData))
+        guard let receiptASN1Container = try findASN1Container(withObjectId: .data, inContainer: asn1Container) else {
+            throw ReceiptReadingError.dataObjectIdentifierMissing
+        }
+        let receipt = try receiptBuilder.build(fromASN1Container: receiptASN1Container)
+        return receipt
+    }
+}
+
+private extension ReceiptParser {
+    func findASN1Container(withObjectId objectId: ASN1ObjectIdentifier,
```

Comment:
this would be reused if we wanted other parts of the receipt, like the signature